Dockerize cavetl projects

slifty opened the pull request:
Docker makes it easy to spin up a working copy of the leie module, and a theoretically working copy of the dmf module.  This sets up the Dockerfilse as well as a more general docker-compose file which coordinates multiple containers at once.

Resolves #11

slifty committed:
Dockerize leie module

Docker makes it easy to spin up a working copy of the leie module.  This
sets up the Dockerfile as well as a more general docker-compose file
(since we will eventually have more than one container associated with
this repository).

Since docker is a container, the flask server also needed to be modified
to listen to non-local traffic.

slifty committed:
Dockerize dmf

There are currently two projects in this repository, and this dockerizes
the second as well. It appears that there are a few config settings that
were not properly reflected in the repository's config.py, so those have
been added.

Issue #11 Put applications into containers

frankduncan reviewed (changes requested):
An update to the top level readme needed for docker instructions.

slifty committed:
Add dockermumentation

Insert some basic instructions on how to run the docker container setup
for this project.

Issue #11 Put applications into containers

Commit:
Update DMF to be dockerable

frankduncan reviewed (approved):
Tested and works after making some addition changes (checked in)

kfogel commented:
Gonna get this merged ASAP.  [Discussion here.](https://chat.opentechstrategies.com/#narrow/stream/6-Provider-Screening/subject/cavetl/near/52248)

kfogel commented on dmf/config.py:
```diff
@@ -4,3 +4,5 @@
 HTTPS_PROXY=None
 HTTP_PROXY=None
+SSN_CLIENT=None
+SSN_MODULEN=None
```
This is a typo -- should be `SSN_MODULE` (without the `N`).  I'll fix.

kfogel replied:
Done in commit b0d7baefd3f.  (It will all get washed away in the move to `dmf/config.py.tmpl`, which is coming in a moment via merge, but I wanted the discrepancy between the old file and the new file to be resolved before that.)

kfogel committed:
Fix typo from commit ca63dd4d05

kfogel committed:
Add newline at end of DMF config file

Just for good form; shouldn't make any functional difference.

kfogel committed:
Merge branch 'master' into 11-dockerize-leie

This mainly meant resolving some text conflicts in INSTALL.md and
accepting the removal of dmf/config.py in favor of dmf/config.py.tmpl.
